src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -759,6 +760,32 @@ bool clang_cpp_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
     break;
   }
 
+  case clang::Stmt::SizeOfPackExprClass:
+  {
+    const clang::SizeOfPackExpr &size_of_pack =
+      static_cast<const clang::SizeOfPackExpr &>(stmt);
+    if (size_of_pack.isValueDependent())
+    {
+      std::ostringstream oss;
+      llvm::raw_os_ostream ross(oss);
+      ross << "Conversion of unsupported value-dependent size-of-pack expr: \"";
+      ross << stmt.getStmtClassName() << "\" to expression"
+           << "\n";
+      stmt.dump(ross, *ASTContext);
+      ross.flush();
+      log_error("{}", oss.str());
+      return true;
+    }
+    else
+    {
+      new_expr = constant_exprt(
+        integer2binary(size_of_pack.getPackLength(), bv_width(size_type())),
+        integer2string(size_of_pack.getPackLength()),
+        size_type());
```

Comment:
size_type() should be fine here, indeed. Alternatively, you could've used something like
```c++
typet type;
if (get_type(size_of_pack.getType(), type))
  return true;
```